pom.xml

```diff
@@ -3587,6 +3605,7 @@
                 <configuration>
                     <additionalJOption>-J-Xmx1536m</additionalJOption>
                     <quiet>true</quiet>
+                    <sourcepath>${project.build.directory}/delombok</sourcepath>
```

Comment:
I'm uncomfortable we change the source path for every module away from the defaults  - we have non java code too. In a number of cases in the past going with non-defaults have caused subtle issues elsewhere in the process

I see some references (google!) to keeping lombok'd code in a separate build tree within the component - this means that components without lombok are not affected *at all* (the delombok process will do nothing), whilst the lombok'd code is appropriate delomboked before the rest of the usual process applies. This seems to cause less changes to the overall build process /filesystem layout, and is really just adding in lombok in a less intrusive way

Even with a couple of components using lombok this seems a more natural fit, but it's particularly important if we want to use further in future

Reply:
I agree, is there a way of organising this so that java builds without Lombok are not effected. Maybe the ' separate build tree option'

Reply:
The sourcepath tag above is inside the config of the javadoc plugin and it indicates the place where the plugin should search for java files in order to generate javadoc. sourceDirectory inside the build tag is left untouched so the java build will not be affected, nor the non java code files. 

Also, taken from Delombok documentation: "To keep any changes to your code style to a minimum, delombok just copies a source file directly without changing any of it if the source file contains no lombok transformations." This means that only the modules that really use Lombok will be touched by the plugin. 
That can also be seen in the console when running a mvn clean install command: [INFO] --- lombok-maven-plugin:1.18.20.0:delombok (default) @ open-metadata-deployment ---
[WARNING] Skipping Delombok; no source to process.

In this case I think there is no overload of the build or risks regarding sources files or javadocs.

Reply:
Thanks for clarifying the location of the source tag - I had missed that! Agreed that's reduced impact

We should also double check that with the plugin in place, that the source jar we push to maven central (or snapshot repo) contains appropriate source for debugging that intellij can use, as well as javadoc. (ie testing of the dev/debug scenario outside core egeria, but using egeria artifacts)

I still think it makes sense to do this in two stages - propose an approach for the small number of modules using lombok, merge, ask people to try it out in their environment (ie make code changes , build, debug), give it a period of time (weeks?) and then a new discussion/proposal on applying to all components (ideally only java ones)

Reply:
now agree we don't need to back off in this way.